File: src/main/java/platform/ecommerce/dto/response/MemberResponse.java
package platform.ecommerce.dto.response;

import lombok.Builder;
import platform.ecommerce.domain.member.MemberRole;
import platform.ecommerce.domain.member.MemberStatus;

import java.time.LocalDateTime;

/**
 * Member response DTO.
 */
@Builder
public record MemberResponse(
        Long id,
        String email,
        String name,
        String phone,
        String profileImage,
        MemberRole role,
        MemberStatus status,
        boolean emailVerified,
        LocalDateTime lastLoginAt,
        LocalDateTime createdAt
) {
}

File: src/main/kotlin/platform/ecommerce/dto/response/MemberResponse.kt
package platform.ecommerce.dto.response

import platform.ecommerce.enums.MemberRole
import platform.ecommerce.enums.MemberStatus
import java.time.Instant
import java.util.*

/**
 * Response DTO for member information.
 * Excludes sensitive data like password hash.
 */
data class MemberResponse(
    val id: UUID,
    val email: String,
    val firstName: String,
    val lastName: String,
    val phone: String?,
    val role: MemberRole,
    val status: MemberStatus,
    val createdAt: Instant,
    val updatedAt: Instant
)
